src/Features/VisualBasic/Portable/ExtractInterface/VisualBasicExtractInterfaceService.vb

```diff
@@ -198,5 +201,44 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExtractInterface
             Dim typeDeclaration = DirectCast(typeNode, TypeBlockSyntax)
             Return typeDeclaration.GetModifiers().Any(Function(m) SyntaxFacts.IsAccessibilityModifier(m.Kind()))
         End Function
+
+        Friend Overrides Function GetSolutionWithSameFileUpdated(
+            solution As Solution,
+            extractedInterfaceSymbol As INamedTypeSymbol,
+            includedMembers As IEnumerable(Of ISymbol),
+            symbolToDeclarationAnnotationMap As Dictionary(Of ISymbol, SyntaxAnnotation),
+            documentIds As List(Of DocumentId),
+            typeNodeAnnotation As SyntaxAnnotation,
+            documentIdWithTypeNode As DocumentId,
+            cancellationToken As CancellationToken) As Solution
+
+            Dim updatedSolution = GetSolutionWithUpdatedOriginalType(solution,
+                                               extractedInterfaceSymbol,
+                                               includedMembers,
+                                               symbolToDeclarationAnnotationMap,
+                                               documentIds,
+                                               typeNodeAnnotation,
+                                               documentIdWithTypeNode,
+                                               cancellationToken)
+
+            Dim document = updatedSolution.GetDocument(documentIdWithTypeNode)
+            Dim originalRoot = document.GetSyntaxRootSynchronously(cancellationToken)
+            Dim typeDeclaration = originalRoot.GetAnnotatedNodes(typeNodeAnnotation).Single()
+
+            document = document.WithSyntaxRoot(originalRoot.TrackNodes(typeDeclaration))
+
+            Dim currentRoot = document.GetSyntaxRootSynchronously(cancellationToken)
+            Dim editor = New SyntaxEditor(currentRoot, VisualBasicSyntaxGenerator.Instance)
+
+            typeDeclaration = currentRoot.GetCurrentNode(typeDeclaration)
+
+            Dim codeGenService = updatedSolution.Workspace.Services.GetLanguageServices(LanguageNames.VisualBasic).GetService(Of ICodeGenerationService)
+            Dim interfaceNode = codeGenService.CreateNamedTypeDeclaration(extractedInterfaceSymbol)
+
+            editor.InsertBefore(typeDeclaration, interfaceNode)
+
+            Dim newRoot = Formatter.Format(editor.GetChangedRoot(), updatedSolution.Workspace)
+            Return updatedSolution.WithDocumentSyntaxRoot(documentIdWithTypeNode, newRoot, PreservationMode.PreserveIdentity)
```

Comment:
it seems like it would be wholely unnecessary to have language specific code for this case.  Why can this code not be shared with the C# impl?